src/main/kotlin/com/personio/synthetics/builder/SyntheticTestBuilder.kt

```diff
@@ -0,0 +1,291 @@
+) {
+    protected var parameters: SyntheticTestParameters
+    protected var options: SyntheticsTestOptions
+    protected var locations: List<String> = defaults.runLocations
+
+    init {
+        parameters = SyntheticTestParameters(
+            message = "",
+            locations = defaults.runLocations,
+            tags = mutableListOf()
+        )
+
+        options = SyntheticsTestOptions()
+            .addDeviceIdsItem(SyntheticsDeviceID.CHROME_LAPTOP_LARGE)
+            .tickEvery(defaults.testFrequency / 1000)
+            .minFailureDuration(defaults.minFailureDuration / 1000)
+            .minLocationFailed(defaults.minLocationFailed)
+            .retry(
+                SyntheticsTestOptionsRetry()
+                    .count(defaults.retryCount)
+                    .interval(defaults.retryInterval)
+            )
+            .monitorOptions(
+                SyntheticsTestOptionsMonitorOptions()
+            )
+    }
+
+    /**
+     * Sets the monitor name for the synthetic test
+     * @param monitorName The monitor name of the test
+     */
+    fun monitorName(monitorName: String) {
+        options.monitorName = monitorName
+    }
+
+    /**
+     * Configures the alert message to be sent when the test fails
+     * @param failureMessage The message that needs to be sent to the configured alert medium upon test failure
+     * @param alertMedium Specify one or more alert mediums. It can be Slack channels, email addresses and so on
+     * For Slack channels, the channel name should be prefixed with "@slack-"
+     * For email address, the email should be prefixed with "@"
+     * Example: Slack channel -> @slack-test_slack_channel
+     * Email -> @firstName.lastName@domain.com
+     */
+    fun alertMessage(failureMessage: String, vararg alertMedium: String) {
+        parameters.message += "${alertMedium.joinToString(" ")} {{#is_alert}} $failureMessage {{/is_alert}} "
+    }
+
+    /**
+     * Configures the recovery message when the test recovers
+     * @param recoveryMessage The message that needs to be sent upon recovery from a failure
+     */
+    fun recoveryMessage(recoveryMessage: String) {
+        parameters.message += " {{#is_recovery}} $recoveryMessage {{/is_recovery}} "
+    }
+
+    /**
+     * Sets the tags for the synthetic test
+     * @param tags List of tags
+     */
+    fun tags(vararg tags: String) {
+        parameters.tags.addAll(tags)
+    }
+
+    /**
+     * Sets the locations for the synthetic test
+     * @param locations List of locations
+     */
+    fun publicLocations(vararg locations: Location) {
+        parameters.locations = locations.map { it.value }
+    }
+
+    /**
+     * Sets the test frequency for the synthetic test
+     * @param frequency The frequency of the test
+     * Allowed test frequency is between 5 minutes and 7 days
+     */
+    fun testFrequency(frequency: Duration) {
+        options.tickEvery = frequency.inWholeSeconds
```

Comment:
Would be nice to set [restrictions](https://github.com/personio/datadog-synthetic-test-support/blob/master/src/main/kotlin/com/personio/synthetics/config/BrowserTestConfig.kt#L41) like on UI